### PR TITLE
Rubicon: Support sending segments to XAPI

### PR DESCRIPTION
**Pull request by snahornyi**

### CODE REVIEW

**Review by guscarreon — APPROVED** [review]
Looks good to me. If anything, a little nitpick in regards to renaming `Taxonomyname` to `TaxonomyName` in line 97 of `rubigon.go`, but I can live with its current name too.

**Review by guscarreon — DISMISSED** [review]
LGTM.

File: adapters/rubicon/rubicon_test.go [hunk 1]
```diff
@@ -1320,6 +1320,79 @@ func TestOpenRTBRequestWithSpecificExtUserEids(t *testing.T) {
 	assert.Contains(t, userExtRPTarget["LIseg"], "999", "No segment with 999 as expected!")
 }
 
+func TestOpenRTBRequestWithTaxonomynameAttribute(t *testing.T) {
```
Comment (SyntaxNode) [inline adapters/rubicon/rubicon_test.go:1323]: I see you updated the `simple-video.json` test to include this new logic. Is this `TestOpenRTBRequestWithTaxonomynameAttribute` test a duplicate, essentially?

Comment (snahornyi) [reply adapters/rubicon/rubicon_test.go:1323]: @SyntaxNode Yes, actually I don't believe even unit tests, so also updated json test, should this update be removed?

Comment (SyntaxNode) [reply adapters/rubicon/rubicon_test.go:1323]: I just care about the json test. The assertions for the json test framework will soon include memory protection checks as a replacement for the near-useless race condition checks.

Comment (snahornyi) [reply adapters/rubicon/rubicon_test.go:1323]: Not clearly understand, what should I do?

Comment (SyntaxNode) [reply adapters/rubicon/rubicon_test.go:1323]: Please delete the `TestOpenRTBRequestWithTaxonomynameAttribute` test and express it as a json file in either the "exemplary" or "supplemental" directories. This is my take on the conversion:

```
{
    "mockBidRequest": {
        "id": "test-request-id",
        "site": {
            "page": "mysite.com"
        },
        "imp": [{
            "id": "test-imp-id",
            "banner": {
                "format": [{
                    "w": 300,
                    "h": 250
                }, {
                    "w": 300,
                    "h": 600
                }]
            },
            "ext": {
                "bidder": {
                    "zoneId": 8394,
                    "siteId": 283282,
                    "accountId": 7891
                }
            }
        }],
        "user": {
            "data": [{
                    "ext": {
                        "taxonomyname": "iab"
                    },
                    "segment": [{
                        "id": "idToCopy"
                    }]
                }, {
                    "ext": {
                        "taxonomyname": "someValue"
                    },
                    "segment": [{
                        "id": "shouldNotBeCopied"
                    }]
                },
                {
                    "ext": {
                        "taxonomyname": "IaB"
                    },
                    "segment": [{
                        "id": "idToCopy2"
                    }]
                },
                {
                    "ext": {
                        "taxonomyname": "wrong iab type"
                    },
                    "segment": [{
                        "id": "shouldNotBeCopied2"
                    }]
                }
            ]
        }
    },
    "httpCalls": [{
        "expectedRequest": {
            "uri": "uri?tk_xint=pbs-test-tracker",
            "body": {
                "id": "test-request-id",
                "site": {
                    "page": "mysite.com",
                    "publisher": {
                        "ext": {
                            "rp": {
                                "account_id": 7891
                            }
                        }
                    },
                    "ext": {
                        "rp": {
                            "site_id": 283282
                        }
                    }
                },
                "imp": [{
                    "id": "test-imp-id",
                    "banner": {
                        "format": [{
                            "w": 300,
                            "h": 250
                        }, {
                            "w": 300,
                            "h": 600
                        }],
                        "ext": {
                            "rp": {
                                "size_id": 15,
                                "alt_size_ids": [10],
                                "mime": "text/html"
                            }
                        }
                    },
                    "ext": {
                        "rp": {
                            "track": {
                                "mint": "",
                                "mint_version": ""
                            },
                            "zone_id": 8394
                        }
                    }
                }],
                "user": {
                    "data": [{
                            "ext": {
                                "taxonomyname": "iab"
                            },
                            "segment": [{
                                "id": "idToCopy"
                            }]
                        }, {
                            "ext": {
                                "taxonomyname": "someValue"
                            },
                            "segment": [{
                                "id": "shouldNotBeCopied"
                            }]
                        },
                        {
                            "ext": {
                                "taxonomyname": "IaB"
                            },
                            "segment": [{
                                "id": "idToCopy2"
                            }]
                        },
                        {
                            "ext": {
                                "taxonomyname": "wrong iab type"
                            },
                            "segment": [{
                                "id": "shouldNotBeCopied2"
                            }]
                        }
                    ],
                    "ext": {
                        "digitrust": null,
                        "rp": {
                            "target": {
                                "iab": [
                                    "idToCopy",
                                    "idToCopy2"
                                ]
                            }
                        }
                    }
                }
            }
        },
        "mockResponse": {
            "status": 200,
            "body": {
                "id": "test-request-id",
                "seatbid": [{
                    "bid": [{
                        "id": "test_bid_id",
                        "impid": "test-imp-id",
                        "price": 0.27543,
                        "adm": "some-test-ad",
                        "cid": "test_cid",
                        "crid": "test_crid",
                        "dealid": "test_dealid",
                        "ext": {
                            "prebid": {
                                "type": "banner"
                            }
                        }
                    }],
                    "seat": "adman"
                }],
                "cur": "USD"
            }
        }
    }],
    "expectedBidResponses": [{
        "currency": "USD",
        "bids": [{
            "bid": {
                "id": "test_bid_id",
                "impid": "test-imp-id",
                "price": 0.27543,
                "adm": "some-test-ad",
                "cid": "test_cid",
                "crid": "test_crid",
                "dealid": "test_dealid",
                "ext": {
                    "prebid": {
                        "type": "banner"
                    }
                }
            },
            "type": "banner"
        }]
    }]
}
```

**Review by SyntaxNode — APPROVED** [review]

**Review by guscarreon — APPROVED** [review]
LGTM

**Pull request merged**